noxfile.py

```diff
@@ -319,6 +319,24 @@ def coverage(session):
     session.run("coverage", "erase")
 
     selected_plugins = select_plugins(session)
+
+    if selected_plugins[0]["name"] == "hydra_ray_launcher":
+        ray_plugin = selected_plugins[0]
+        session.run(
+            "pip",
+            "install",
+            "-e",
+            os.path.join("plugins", ray_plugin["path"]),
+            silent=SILENT,
+        )
+        session.run("coverage", "erase", env=coverage_env)
+        session.chdir(os.path.join("plugins", ray_plugin["path"]))
+        cov_args = ["coverage", "run", "--append", "-m"]
+        cov_args.extend(pytest_args(session))
+        session.run(*cov_args, silent=SILENT, env=coverage_env)
+        session.chdir(BASE)
+        selected_plugins = selected_plugins[1:]
+
```

Comment:
1. If all this is doing is changing the order, why not just manipulate selected_plugins and let the loop run normally?
2. Add a comment explaining why this is needed.

Reply:
before this change the test first installs all the plugin and then run all the tests. The goal of the change is to install ray launcher first and then finish the test before we install any other plugin. 
(to avoid pickling issue)